### PR TITLE
MINOR: only log first exception in RecordCollectorImpl producer callback

### PR DESCRIPTION
**Pull request by dguy**

### CODE REVIEW

**Comment by dguy** [comment]
@guozhangwang @enothereska - i noticed this was creating a lot of noise in the logs i'm trying to debug. There is no point in logging the exception every time

File: streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java [hunk 1]
```diff
@@ -96,8 +96,10 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
                             TopicPartition tp = new TopicPartition(metadata.topic(), metadata.partition());
                             offsets.put(tp, metadata.offset());
                         } else {
-                            sendException = exception;
-                            log.error("{} Error sending record to topic {}. No more offsets will be recorded for this task and the exception will eventually be thrown", logPrefix, topic, exception);
+                            if (sendException == null) {
```
Comment (guozhangwang) [inline streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java:99]: We are effectively changing the behavior to report the first encountered exception to the last one here. Is that OK?

Comment (dguy) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java:99]: we are changing it so it just reports the first exception. Previously it was reporting all of them, which for a batch of records was the same message how ever many times. In the case i was looking at it was a couple of hundred messages.

Comment (guozhangwang) [reply streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java:99]: Fair enough.

**Comment by guozhangwang** [comment]
LGTM and merged to trunk/0.10.2

**Pull request closed without merging**